packages/venia-concept/src/util/__tests__/appendOptionsToPayload.spec.js

```diff
@@ -0,0 +1,51 @@
+import appendOptionsToPayload from '../appendOptionsToPayload';
+
+const sampleItem = {
+    sku: 'PARENT',
+    configurable_options: [
+        {
+            attribute_code: 'first_attribute',
+            attribute_id: '1',
+            id: 3,
+            values: [
+                {
+                    value_index: 1
+                }
+            ]
+        },
+        {
+            attribute_code: 'second_attribute',
+            attribute_id: '2',
+            id: 6,
+            values: [
+                {
+                    value_index: 2
+                }
+            ]
+        }
+    ],
+    variants: [
+        {
+            product: {
+                first_attribute: 1,
+                second_attribute: 2,
+                id: 7
+            }
+        }
+    ]
+};
+
+test('appends the given options to the payload object', () => {
+    const payload = {
+        item: sampleItem,
+        quantity: 1,
+        productType: 'ConfigurableProduct'
+    };
+    const optionSelections = new Map();
+    optionSelections.set('1', 1);
+    optionSelections.set('2', 2);
+
+    expect(Array.isArray(payload.options)).toBe(false);
+    appendOptionsToPayload(payload, optionSelections);
+    expect(Array.isArray(payload.options)).toBe(true);
+});
```

Comment:
I've added a few new tests. For the case that a selected variant is not found, the function fails and throws a `TypeError`. Should this behave differently?

Reply:
I think it would be useful to have it return the original `payload` in that case. What do you think?

This will let us catch the error and warn about it too, which will be nice.

Reply:
I like that idea. The unchanged `payload` is returned with an invalid selection now, but this case is not currently handled by the calling components. Do you think handling the error and displaying a message should be part of this PR?

Reply:
No, we're missing a lot of that app-wide and I'm trying to get this in before release, so we can add that later if need be 👍